c7n/filters/core.py

```diff
@@ -1018,3 +1020,142 @@ def reorder(self, items, key=None):
+            filters:
+              - type: list-item
+                key: IpPermissions
+                attrs:
+                  - type: value
+                    key: IpRanges[].CidrIp
+                    value: '0.0.0.0/0'
+                    op: in
+                    value_type: swap
+                  - type: value
+                    key: FromPort
+                    value: 22
+                  - type: value
+                    key: ToPort
+                    value: 22
+          - name: find-task-def-not-using-registry
+            resource: aws.ecs-task-definition
+            filters:
+              - not:
+                - type: list-item
+                  key: containerDefinitions
+                  attrs:
+                    - not:
+                      - type: value
+                        key: image
+                        value: "${account_id}.dkr.ecr.us-east-2.amazonaws.com.*"
+                        op: regex
+    """
+
+    def _get_schema():
+        base_filters = [
+            {'$ref': '#/definitions/filters/event'},
+            {'$ref': '#/definitions/filters/reduce'},
+            {'$ref': '#/definitions/filters/value'},
+            {'$ref': '#/definitions/filters/valuekv'},
+            {
+                'enum': [
+                    'value',
+                    'or',
+                    'and',
+                    'not',
+                    'event',
+                    'reduce',
+                ]
+            }
+        ]
+        any_of = []
+        any_of.extend(base_filters)
+
+        for i in ('and', 'or', 'not',):
+            any_of.append(
+                {
+                    'additional_properties': False,
+                    'properties': {
+                        i: {
+                            'type': 'array',
+                            'items': {
+                                'anyOf': base_filters
+                            }
+                        }
+                    },
+                    'type': 'object'
+                }
+            )
+
+        schema = {
+            'items': {
+                'anyOf': any_of
+            },
+            'type': 'array',
+        }
+        return schema
+
+    schema = type_schema(
+        'list-item',
+        key={'type': 'string'},
+        attrs=_get_schema()
+    )
+
+    def process(self, resources, event=None):
+        compiled = jmespath.compile(self.data['key'])
+        result = []
+        frm = ListItemResourceManager(self.manager.ctx, data={'filters': self.data['attrs']})
+        for r in resources:
+            list_values = compiled.search(r)
+            if not list_values:
+                continue
+            if not isinstance(list_values, list):
+                raise PolicyExecutionError(f'{list_values} is not a list')
+            for idx, list_value in enumerate(list_values):
+                list_value['c7n:_id'] = idx
+            resources = frm.filter_resources(list_values, event)
```

Comment:
bug: this is redefining the resources variable that got passed into the method and that is currently being iterated over, likely needs to be a different variable name, Ala item_resources